tests/StorageCacheTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Tests\BlackBox;
+
+use BlackBox\StorageCache;
+
+/**
+ * @covers \BlackBox\StorageCache
+ *
+ * @author Carlos Lombarte <lombartec@gmail.com>
+ */
+class StorageCacheTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+    * Subject under test.
+    *
+    * @var StorageCache
+    */
+    protected $storage;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject
+     */
+    private $sourceStorageMock;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject
+     */
+    private $storageCacheMock;
+
+    public function setUp()
+    {
+        $this->sourceStorageMock    = $this->getMock(
+            'BlackBox\Storage',
+            ['get', 'set', 'add', 'remove', 'getIterator']
+        );
+        $this->storageCacheMock     = $this->getMock('BlackBox\Storage');
+        $this->storage              = new StorageCache($this->sourceStorageMock, $this->storageCacheMock);
+    }
+
+    public function tearDown()
+    {
+        unset(
+            $this->sourceStorageMock,
+            $this->storageCacheMock,
+            $this->storage
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function gets_data_directly_from_cache()
+    {
+        $cachedData = 'cache';
+
+        $this->storageCacheMock->expects($this->once())
+        ->method('get')
```

Comment:
maybe here add a test `->with(25)` to make sure the cache is called with the right ID.